public/components/common/hocs/error-boundary/with-error-boundary.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import log from 'loglevel';
+import { EuiEmptyPrompt } from '@elastic/eui';
+
+export default class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { error: null, errorInfo: null };
+  }
+
+  componentDidCatch = (error, errorInfo) => catchFunc(error, errorInfo, this);
+
+  render() {
+    // @ts-ignore
+    if (this.state.errorInfo) {
+      return handleError(this);
+    }
+    return this.props.children;
+  }
+}
+
+export const withErrorBoundary = (WrappedComponent) =>
+  class extends Component {
+    constructor(props) {
+      super(props);
+      this.state = { error: null, errorInfo: null };
+    }
+
+    componentDidCatch = (error, errorInfo) => {
+      catchFunc(error, errorInfo, this);
+    };
+
+    render() {
+      debugger;
+      // @ts-ignore
+      if (this.state.errorInfo) {
+        return handleError(this);
+      }
+      return <WrappedComponent {...this.props} />;
+    }
+  };
```

Comment:
I think that the HOC could be refactored to:
```ts
export const withErrorBoundary = WrappedComponent => props => (
  <ErrorBoundary>
    <WrappedComponent {...props} />
  </ErrorBoundary>
)
```

If the refactoring is possible, I would move the `catchFunc`  and `handleError` functions as components methods.

Reply:
Great!
Maybe we can use [this](https://www.npmjs.com/package/react-error-boundary)